README.md

```diff
@@ -0,0 +1,117 @@
+# specfile
+
+Python library for parsing and manipulating RPM spec files. Main focus is on modifying existing spec files, any change should result in a minimal diff.
+
+## Caveats
+
+### RPM macros
+
+specfile uses RPM for parsing spec files and macro expansion. Unfortunately, macros are always stored in a global context, which poses a problem for multiple instances of Specfile.
+
+## Examples and use cases
+
+### Instantiating
+
+```python
+# using an absolute path
+specfile = Specfile('/tmp/test.spec')
+# using a relative path and a different sourcedir
+specfile = Specfile('test.spec', sourcedir='/tmp/sources')
+```
+
+### Saving changes
+
+```python
+# enabling autosave
+specfile = Specfile('test.spec', autosave=True)
+
+# no autosave (the default)
+specfile = Specfile('test.spec')
+...
+# saving explicitly
+specfile.save()
+```
+
+## Low-level manipulation
+
+```python
+# replacing the content of a section
+specfile.sections.prep = ['%autosetup -p1']
+# removing a section
+del specfile.sections.changelog
+# swapping two sections
+specfile.sections[1], specfile.sections[2] = specfile.sections[2], specfile.sections[1]
+# accessing a section with arguments
+print(getattr(specfile.sections, 'package devel'))
+# inserting a line into a section
+specfile.sections.build.insert(0, 'export VERBOSE=1')
+
+# copying a section from one specfile to another
+specfile2.sections.changelog[:] = specfile1.sections.changelog
+```
+
+## Mid-level - tags
+
+```python
+# accessing tags in preamble
+with specfile.tags() as tags:
+    # name of the first tag
+    print(tags[0].name)
+    # expanded value of the first tag
+    print(tags[0].value)
+    # raw value of the first tag
+    print(tags[0].raw_value)
+    # comment associated with the first tag
+    print(tags[0].comment)
+    # value of a tag by name
+    print(tags.url)
+
+# accessing tags in subpackages
+with specfile.tags('package devel') as tags:
+    print(tags.requires)
+```
+
+## High-level
+
+### Version and release
+
+```python
+# getting version and release
+print(specfile.version)
+print(specfile.release)
+# setting version and release
```

Comment:
I'd mention that this sets it as raw content. On one hand, this is not official documentation, however we should make the expectations clear even here.

Reply:
I agree, but then it's necessary to explain what raw means in this context.